lib/logstash/outputs/dynatrace.rb

```diff
@@ -106,28 +106,23 @@ def run
         end
       end
 
-      class BatchSerializer
+      class Batcher
         def initialize(max_batch_size)
           @max_batch_size = max_batch_size
           @batch_events_size = 0
           @serialized_events = []
         end
 
-        def offer(event)
-          serialized_event = LogStash::Json.dump(event.to_hash)
-          if batch_size + serialized_event.length + (@serialized_events.length.positive? ? 1 : 0) > @max_batch_size
-            return false
-          end
+        def offer(serialized_event)
+          # 2 square brackets, the length of all previously serialized strings, commas, and the current event size
+          batch_size_bytes = 2 + @batch_events_size + @serialized_events.length + serialized_event.length
```

Comment:
Should this be serialized_events.length-1 since there is no comma after the last entry?

Reply:
this is the count of previously serialized events, which is actually the correct count of commas when the new event is added. The `-1`s and `+1`s of the previous version cancelled each other out.

1. first event is added. serialized events is 0 and 0 commas are in the output
2. second event is added. serialized events is 1 and 1 comma is added
3. and so on